decidim-proposals/app/controllers/concerns/decidim/proposals/orderable.rb

```diff
@@ -16,18 +16,51 @@ module Orderable
         # Available orders based on enabled settings
         def available_orders
           @available_orders ||= begin
-            available_orders = %w(random recent)
-            available_orders << "most_voted" if most_voted_order_available?
-            available_orders << "most_endorsed" if current_settings.endorsements_enabled?
-            available_orders << "most_commented" if component_settings.comments_enabled?
-            available_orders << "most_followed" << "with_more_authors"
+            available_orders = [default_order]
+            available_orders.concat(possible_orders.reject { |order| order == default_order })
             available_orders
           end
         end
 
+        def possible_orders
+          @possible_orders ||= begin
+            possible_orders = %w(random recent)
+            possible_orders << "most_voted" if most_voted_order_available?
+            possible_orders << "most_endorsed" if current_settings.endorsements_enabled?
+            possible_orders << "most_commented" if component_settings.comments_enabled?
+            possible_orders << "most_followed" << "with_more_authors"
+            possible_orders
+          end
+        end
+
         def default_order
+          @default_order ||= begin
+            if active_step
+              step_default_sort_order = current_component[:settings].dig("steps", active_step.id.to_s, "default_sort_order")
```

Comment:
We should utilize the `current_settings` and `component_settings` utility methods instead of fetching them manually:
https://github.com/decidim/decidim/blob/20b6cb42bf0363046b46e2c307817c8aec724e78/decidim-core/app/controllers/concerns/decidim/settings.rb#L14-L20

So something like:

```ruby
current_settings.default_sort_order || component_settings.default_sort_order
```